clients/src/main/java/org/apache/kafka/clients/consumer/internals/Heartbeat.java

```diff
@@ -42,67 +45,60 @@ public Heartbeat(int sessionTimeoutMs,
         this.heartbeatIntervalMs = heartbeatIntervalMs;
         this.maxPollIntervalMs = maxPollIntervalMs;
         this.retryBackoffMs = retryBackoffMs;
+        this.heartbeatTimer = time.timer(heartbeatIntervalMs);
+        this.sessionTimer = time.timer(sessionTimeoutMs);
+        this.pollTimer = time.timer(maxPollIntervalMs);
     }
 
-    public void poll(long now) {
-        this.lastPoll = now;
+    public void poll() {
+        pollTimer.reset(maxPollIntervalMs);
```

Comment:
I'm a bit concerned about the perf implication here: we call an additional `time.millis` every time with `consumer.poll()` now, could we still pass in `now` and add a `reset` function that accepts the currentTime like the overloaded `update` as well?

Reply:
I think we're probably still ahead in terms of total calls to `time.milliseconds`, but I can try to remove this one also.